src/planwise/component/maps.clj

```diff
@@ -44,26 +48,28 @@
 
 (defn demand-map
   [service region-id facilities]
-  (let [polygons (filter :polygon-id facilities)
-        polygons-with-capacities (->> polygons
-                                    (map (juxt
-                                          #(isochrones-path service region-id "/" (:polygon-id %) ".tif")
-                                          #(str (capacity-for service %))))
-                                    (flatten))
-        map-key  (demand-map-key region-id polygons-with-capacities)
-        response (apply run-external
-                    (:runner service)
-                    :bin
-                    "calculate-demand"
-                    (demands-path service map-key ".tif")
-                    (populations-path service region-id ".tif")
-                    (vec polygons-with-capacities))
-        unsatisfied-count (-> response
-                            (str/trim-newline)
-                            (str/trim)
-                            (Integer.))]
-      {:map-key map-key,
-       :unsatisfied-count unsatisfied-count}))
+  (if-not (calculate-demand? service)
+    {}
+    (let [polygons (filter :polygon-id facilities)
```

Comment:
Why? `nil` is equivalent in 99% of the cases (only exception I can think of is `map?`). The only use I could find of this function is immediately followed by an `assoc` which works as intended with `nil`.